File: example/java/jimmer-cloud/book-service/src/main/java/org/babyfish/jimmer/example/cloud/book/cfg/JimmerConfig.java
package org.babyfish.jimmer.example.cloud.book.cfg;

import org.springframework.cloud.client.loadbalancer.LoadBalanced;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestTemplate;

@Configuration
public class JimmerConfig {

    // Jimmer uses this load balanced client to fetch data across microservices
    @LoadBalanced
    @Bean
    public RestTemplate restTemplate() {
        return new RestTemplate();
    }
}

File: example/kotlin/jimmer-cloud-kt/book-service/src/main/kotlin/org/babyfish/jimmer/example/cloud/kt/book/cfg/JimmerConfig.kt
package org.babyfish.jimmer.example.cloud.kt.book.cfg

import org.springframework.cloud.client.loadbalancer.LoadBalanced
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestTemplate

@Configuration
class JimmerConfig {

    // Jimmer uses this load balanced client to fetch data across microservices
    @LoadBalanced
    @Bean
    fun restTemplate() = RestTemplate()
}